tests/clvm/test_chialisp_deserialization.py

```diff
@@ -0,0 +1,111 @@
+                (size >> 24) & 0xff,
+                (size >> 16) & 0xff,
+                (size >> 8) & 0xff,
+                (size >> 0) & 0xff,
+            ]
+        )
+    else:
+        size_blob = bytes(
+            [
+                0xfc | ((size >> 40) & 0xff),
+                (size >> 32) & 0xff,
+                (size >> 24) & 0xff,
+                (size >> 16) & 0xff,
+                (size >> 8) & 0xff,
+                (size >> 0) & 0xff,
+            ]
+        )
+    extra_str = "01" * 1000
+    return size_blob.hex() + extra_str
+
+
+class TestClvmNativeDeserialization(TestCase):
+    """
+    Test clvm deserialization done from within the clvm
+    """
+
+    def test_deserialization_simple_list(self):
+        # ("hello" "friend")
+        b = hexstr_to_bytes("ff8568656c6c6fff86667269656e6480")
+        cost, output = DESERIALIZE_MOD.run_with_cost([b])
+        print(cost, output)
+        prog = Program.to(output)
+        assert prog == Program.from_bytes(b)
+
+    def test_deserialization_password_coin(self):
+        # (i (= (sha256 2) (q 0x2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824)) (c (q 51) (c 5 (c (q 100) (q ())))) (q "wrong password"))  # noqa
+        b = hexstr_to_bytes("ff04ffff0affff0bff0280ffff01ffa02cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b98248080ffff05ffff01ff3380ffff05ff05ffff05ffff01ff6480ffff01ff8080808080ffff01ff8e77726f6e672070617373776f72648080")  # noqa
+        cost, output = DESERIALIZE_MOD.run_with_cost([b])
+        print(cost, output)
+        prog = Program.to(output)
+        assert prog == Program.from_bytes(b)
+
+    def test_deserialization_large_numbers(self):
+        # '(99999999999999999999999999999999999999999999999999999999999999999 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF -99999999999999999999999999999999999999999999999999999999999999999999999999999)'  # noqa
+        b = hexstr_to_bytes("ff9c00f316271c7fc3908a8bef464e3945ef7a253609ffffffffffffffffffb00fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffa1ff22ea0179500526edb610f148ec0c614155678491902d6000000000000000000180")  # noqa
+        cost, output = DESERIALIZE_MOD.run_with_cost([b])
+        print(cost, output)
+        prog = Program.to(output)
+        assert prog == Program.from_bytes(b)
+
+    def test_overflow_atoms(self):
+        b = hexstr_to_bytes(serialized_atom_overflow(0xffffffff))
+        try:
+            cost, output = DESERIALIZE_MOD.run_with_cost([b])
+        except Exception:
+            assert True
+        else:
+            assert False
+        b = hexstr_to_bytes(serialized_atom_overflow(0x3ffffffff))
+        try:
+            cost, output = DESERIALIZE_MOD.run_with_cost([b])
+        except Exception:
+            assert True
+        else:
+            assert False
+        b = hexstr_to_bytes(serialized_atom_overflow(0xffffffffff))
```

Comment:
This construct here is a lot more clear for exceptions: https://docs.python.org/3/library/unittest.html#basic-example

Example:

```
    def test_split(self):
        with self.assertRaises(TypeError):
            s.split(2)
```